Code/Framework/AzFramework/AzFramework/Input/Mappings/InputMapping.h

```diff
@@ -26,6 +27,111 @@ namespace AzFramework
+            // Reflection
+            static void Reflect(AZ::ReflectContext* context);
+
+            ////////////////////////////////////////////////////////////////////////////////////////////
+            //! Constructor
+            InputChannelNameFilteredByDeviceType();
+
+            ////////////////////////////////////////////////////////////////////////////////////////////
+            //! Destructor
+            virtual ~InputChannelNameFilteredByDeviceType() = default;
+
+            ////////////////////////////////////////////////////////////////////////////////////////////
+            //! Get the currently selected input device type.
+            //! \return Currently selected input device type.
+            inline const AZStd::string& GetInputDeviceType() const { return m_inputDeviceType; }
+
+            ////////////////////////////////////////////////////////////////////////////////////////////
+            //! Get the currently selected input channel name.
+            //! \return Currently selected input channel name.
+            inline const AZStd::string& GetInputChannelName() const { return m_inputChannelName; }
+
+        protected:
+            ////////////////////////////////////////////////////////////////////////////////////////////
+            //! Called when an input device type is selected.
+            //! \return The AZ::Edit::PropertyRefreshLevels to apply to the property tree view.
+            virtual AZ::Crc32 OnInputDeviceTypeSelected();
+
+            ////////////////////////////////////////////////////////////////////////////////////////////
+            //! Get the name label override to display.
+            //! \return Name label override to display.
+            virtual AZStd::string GetNameLabelOverride() const;
+
+            ////////////////////////////////////////////////////////////////////////////////////////////
+            //! Get the valid input device types for this input mapping.
+            //! \return Valid input device types for this input mapping.
+            virtual const AZStd::vector<AZStd::string> GetValidInputDeviceTypes() const;
+
+            ////////////////////////////////////////////////////////////////////////////////////////////
+            //! Get the valid input channel names for this input mapping given the selected device type.
+            //! \return Valid input channel names for this input mapping given the selected device type.
+            virtual const AZStd::vector<AZStd::string> GetValidInputChannelNamesBySelectedDevice() const;
+
+        private:
+            ////////////////////////////////////////////////////////////////////////////////////////////
+            // Variables
+            AZStd::string m_inputDeviceType = "";   //!< The currently selected input device type.
+            AZStd::string m_inputChannelName = "";  //!< The currently selected input channel name.
```

Comment:
This is a pessimization.
The string class default constructor initializes to an empty string.

By initializing a string to "" that forces a `strlen` over the supplied literal to be invoked, which is slightly slower than **doing nothing**

Reply:
I just learnt a new word: pessimization :)

Will update.